kernel/rros/c_trace.c

```diff
@@ -0,0 +1,326 @@
+#define CREATE_TRACE_POINTS
+#include <trace/events/rros.h>
+#include <uapi/linux/uio.h>
+
+#define copy_on_stack(dst,src) \
+char dst[src.iov_len+1]; \
+strncpy(dst,src.iov_base,src.iov_len); \
+dst[src.iov_len] = 0;
+
+void rust_helper_trace_rros_schedule(unsigned long flags,unsigned long local_flags)
+{
+        trace_rros_schedule(flags,local_flags);
+}
+EXPORT_SYMBOL_GPL(rust_helper_trace_rros_schedule);
+
+void rust_helper_trace_rros_reschedule_ipi(unsigned long flags,unsigned long local_flags)
+{
+        trace_rros_reschedule_ipi(flags,local_flags);
+}
+EXPORT_SYMBOL_GPL(rust_helper_trace_rros_reschedule_ipi);
+
+void rust_helper_trace_rros_pick_thread(struct iovec  name_struct,pid_t next_pid)
+{
+        copy_on_stack(name,name_struct);
+        trace_rros_pick_thread(name,next_pid);
+}
+EXPORT_SYMBOL_GPL(rust_helper_trace_rros_pick_thread);
+
+void rust_helper_trace_rros_switch_context(struct iovec prev_name_struct,struct iovec next_name_struct,pid_t prev_pid,int prev_prio,u32 prev_state,pid_t next_pid,int next_prio)
```

Comment:
We use the struct `iovec` to stand for a string with a fixed length. It might be ambiguous for someone who use the API for the first time. I think leaving a comment here will be better.